app/executor.php

```diff
@@ -116,230 +130,184 @@ function logError(Throwable $error, string $action, Utopia\Route $route = null)
 
-        if ($activeRuntimes->exists($container)) {
+        if ($activeRuntimes->exists($runtimeId)) {
             throw new Exception('Runtime already exists.', 409);
         }
 
-        $build = [];
-        $buildId = '';
-        $buildStdout = '';
-        $buildStderr = '';
-        $buildStart = \time();
-        $buildEnd = 0;
+        $container = [];
+        $containerId = '';
+        $stdout = '';
+        $stderr = '';
+        $startTime = \time();
+        $endTime = 0;
 
         try {
-            Console::info('Building runtime with ID : ' . $runtimeId);
+            Console::info('Building container : ' . $runtimeId);
             /** 
              * Temporary file paths in the executor 
              */
             $tmpSource = "/tmp/$runtimeId/code.tar.gz";
-            $tmpBuildDir = "/tmp/$runtimeId/builds";
             $tmpBuild = "/tmp/$runtimeId/builds/code.tar.gz";
 
             /**
              * Copy code files from source to a temporary location on the executor
              */
-            $device = new Local($destination);
+            $device = new Local();
             $buffer = $device->read($source);
             if(!$device->write($tmpSource, $buffer)) {
                 throw new Exception('Failed to copy source code to temporary directory', 500);
             };
 
             /**
-             * Create a temporary folder to store builds
+             * Create the mount folder
              */
-            if (!\file_exists($tmpBuildDir)) {
-                if (!@\mkdir($tmpBuildDir, 0755, true)) {
-                    throw new Exception("Can't create directory : $tmpBuildDir", 500);
+            if (!\file_exists(\dirname($tmpBuild))) {
+                if (!@\mkdir(\dirname($tmpBuild), 0755, true)) {
+                    throw new Exception("Failed to create temporary directory", 500);
                 }
             }
 
             /**
              * Create container
              */
             $orchestration = $orchestrationPool->get();
-            $container = 'b-' . $runtimeId;
+            $secret = \bin2hex(\random_bytes(16));
+            $vars = \array_merge($vars, [
+                'INTERNAL_RUNTIME_KEY' => $secret,
+                'INTERNAL_RUNTIME_ENTRYPOINT' => $entrypoint,
+            ]);
             $vars = array_map(fn ($v) => strval($v), $vars);
             $orchestration
                 ->setCpus(App::getEnv('_APP_FUNCTIONS_CPUS', 1))
                 ->setMemory(App::getEnv('_APP_FUNCTIONS_MEMORY', 256))
                 ->setSwap(App::getEnv('_APP_FUNCTIONS_MEMORY_SWAP', 256));
             
-            $buildId = $orchestration->run(
+            $entrypoint = empty($commands) ? [] : [
+                'tail',
+                '-f',
+                '/dev/null'
+            ];
```

Comment:
don't understand what we are doing here?

Reply:
Okay so ..

In each runtime, we have an entrypoint command that runs `start.sh`

This script expects the build output to be present in /tmp/code.tar.gs
https://github.com/open-runtimes/open-runtimes/blob/main/runtimes/ruby-3.1/start.sh#L2

If the built code files are not present, the entry point command fails and the container shuts down. 

So what were doing here is essentially this

If there are build commands to be executed, keep the container active using `tail -f ...` else the built files are already preset so we dont need it.